### DISCUSSION TITLE
Help with a sample script

### DISCUSSION THREAD

**Discussion by thaihugo** [jcorporation/mympd-scripts#12] [Q&A] [answered]
Hi,

I'm not an expert at Lua, but I'm trying to scripts things using the new ability. So far, I can launch normal functions without parameter. However, I would appreciate  some sample script or help specially  on this topic:

How can I set a mympd parameter using MYMPD_API_SETTINGS_SET

For example, I want to write something like:
mympd_api("MYMPD_API_SETTINGS_SET", "volume","40") 
or 
mympd_api("MYMPD_API_SETTINGS_SET", "jukebox_unique_tag","Album")
mympd_api("MYMPD_API_SETTINGS_SET", "jukebox_last_played:20")

I'm sorry for this basic question, but it would actually make me jumpstart this whole scripting learning.

Thanks

**Comment by jcorporation** [comment, marked as answer]
Cool, someone uses this new possibility! I am glad to help you. I am thinking about a small new repository with myMPD scripts.

Some examples: https://github.com/jcorporation/myMPD/wiki/Scripting
The API documentation (wip): https://github.com/jcorporation/myMPD/wiki/API (I recently updated it)

You use the false function to set the volume:
`mympd_api("MPD_API_PLAYER_VOLUME_SET", "volume", 40)`

To set other settings (settings in the API are mostly camelCase in contrast to files in the state directory):
`mympd_api("MYMPD_API_SETTINGS_SET", "jukeboxUniqueTag", "Album")`
`mympd_api("MYMPD_API_SETTINGS_SET", "jukeboxLastPlayed", 20)`

You can also set this two settings in one call:
`mympd_api("MYMPD_API_SETTINGS_SET", "jukeboxUniqueTag", "Album", "jukeboxLastPlayed", 20)`

myMPD 6.5.1 includes some bug fixes for scripting.

**Comment by thaihugo** [comment]
Thank you. 6.5.1 did the trick. I thought mympd crashes where due to a bad syntax, but now I see the lua code is working in the newest dev version.
I can fully automate my playlists changes with a crontab, so I can hook that to different system events.

**Comment by jcorporation** [comment]
Great! Please report crashes always!
